eth/api.go

```diff
@@ -1168,7 +1168,7 @@ func (tx *Tx) UnmarshalJSON(b []byte) (err error) {
 		tx.GasLimit = rpc.NewHexNumber(0)
 	}
 	if tx.GasPrice == nil {
-		tx.GasPrice = rpc.NewHexNumber(int64(50000000000))
+		tx.GasPrice = rpc.NewHexNumber(0)
```

Comment:
It is used in the resend RPC method. I agree in this case that a default value isn't a good solution.

I believe the `Resend` method was a contribution from the community and its purpose is to replace an existing transaction with one with a different (higher) gas price. This is convenient when the old transaction isn't processed due to a too low gas price. Without some custom signing tricks this could lock an account since transactions are executed in nonce order. Setting a default gas price is in this case is imho wrong.

The best solution is probably to throw an error since resending a transaction without an explicit new gas price doesn't make sense.